packages/paste-core/components/visual-picker/src/VisualPickerRadio.tsx

```diff
@@ -0,0 +1,139 @@
+import {BaseRadioCheckboxControl, BaseRadioCheckboxLabel} from '@twilio-paste/base-radio-checkbox';
+import {ScreenReaderOnly} from '@twilio-paste/screen-reader-only';
+import {useMergeRefs} from '@twilio-paste/utils';
+import {type BorderColor} from '@twilio-paste/style-props';
+
+import {VisualPickerRadioContext} from './VisualPickerContext';
+import {type VisualPickerRadioProps} from './types';
+
+export const VisualPickerRadio = React.forwardRef<HTMLInputElement, VisualPickerRadioProps>(
+  (
+    {id, name, element = 'VISUAL_PICKER_RADIO', value, disabled, hasError, onChange, children, labelText, ...props},
+    ref
+  ) => {
+    const [isHovering, setIsHovering] = React.useState(false);
+
+    const visualPickerRadioGroupContext = React.useContext(VisualPickerRadioContext);
+    const helpTextId = useUID();
+    const radioId = id ? id : useUID();
+    const labelId = useUID();
+
+    const handleChange = React.useCallback(
+      (event: React.ChangeEvent<HTMLInputElement>): void => {
+        if (onChange != null) {
+          onChange(event);
+        } else {
+          visualPickerRadioGroupContext.onChange(event);
+        }
+      },
+      [onChange, visualPickerRadioGroupContext.onChange]
+    );
+
+    const state: HiddenRadioState = {
+      name: visualPickerRadioGroupContext.name || name || visualPickerRadioGroupContext.name,
+      disabled: disabled != null ? disabled : visualPickerRadioGroupContext.groupIsDisabled,
+      hasError: hasError != null ? hasError : visualPickerRadioGroupContext.groupHasError,
+      checked: visualPickerRadioGroupContext.value === value,
+    };
+
+    const {groupHasError, groupIsDisabled} = visualPickerRadioGroupContext;
+
+    const internalRef = React.useRef<HTMLInputElement>();
+    const mergedRef = useMergeRefs(internalRef, ref) as React.Ref<HTMLInputElement>;
+
+    let pickerBorderColor = 'colorBorderWeak' as BorderColor;
+    if (disabled || groupIsDisabled) pickerBorderColor = 'colorBorderWeak';
+    else if (hasError || groupHasError) pickerBorderColor = 'colorBorderError' as BorderColor;
+    else if (state.checked === true) pickerBorderColor = 'colorBorderPrimary' as BorderColor;
+
+    return (
+      <Box
+        element={element}
+        position="relative"
+        display="inline-flex"
+        alignItems={visualPickerRadioGroupContext.orientation === 'vertical' ? 'center' : 'flex-start'}
+        flexDirection="row"
+        backgroundColor={disabled || groupIsDisabled ? 'colorBackgroundWeak' : undefined}
+        borderStyle="solid"
+        borderColor={pickerBorderColor}
+        borderWidth="borderWidth10"
+        borderRadius="borderRadius30"
+        padding="space50"
+        onMouseEnter={() => setIsHovering(true)}
+        onMouseLeave={() => setIsHovering(false)}
+        onClick={(e) => {
+          internalRef.current?.click();
+          internalRef.current?.focus();
+          e.preventDefault();
+          e.stopPropagation();
+        }}
```

Comment:
sorry, one more decision from today:

todo: Keep the hover border colors for radio when it's selected at colorBorderError and colorBorderPrimary.

i.e., there should be no change from the default color on hover since you can't unselect a radio. (technically, we'd want this for the Radio input dot, too, but they're sharing the same styles with Checkbox input, so... shrug)